XenAdmin/TabPages/GeneralTabPage.cs

```diff
@@ -179,13 +191,21 @@ public IXenObject XenObject
             }
         }
 
-        private void SetupAnStartLicenseStatus(IXenObject xo)
+        private void SetupAndStartLicenseStatus(IXenObject xo)
         {
+            System.Diagnostics.Debug.Assert(xo is Host || xo is Pool);
+
             licenseStatus = new LicenseStatus(xo);
             licenseStatus.ItemUpdated += licenseStatus_ItemUpdated;
             licenseStatus.BeginUpdate();
         }
 
+        private void UnregisterLicenseStatusUpdater()

```

Comment:
I think we should combine SetupAndStartLicenseStatus and UnregisterLicenseStatusUpdater into one function to do the unregistering and registering, so that they're obviously next to each other in the code flow.